compiler/resolution/functionResolution.cpp

```diff
@@ -8928,10 +8928,9 @@ static void insertUnrefForArrayReturn(FnSymbol* fn) {
           call->insertAtTail(unrefCall);
           FnSymbol* unrefFn = resolveNormalCall(unrefCall);
           resolveFns(unrefFn);
-
-          if (unrefFn->retType == tmp->typeInfo()) {
-            // If the types are equal, we must be dealing with a non-view
-            // array, so we can remove the useless unref call.
+          if (unrefFn->where == NULL) {
```

Comment:
@benharsh: Would using a pragma/flag be a more principled way of distinguishing the array view overload of chpl__unref() from the others?  (assuming I'm understanding this properly).  It seems we could have other chpl__unref()s down the line that will also have where clauses over time?